lib/eventFactory.js

```diff
@@ -2,9 +2,61 @@
 
 const BaseFactory = require('./baseFactory');
 const Event = require('./event');
+const jobConfigSchema = require('screwdriver-data-schema').config.job;
+const joi = require('joi');
 
 let instance;
 
+/**
+ * Create builds associated with this event
+ * For example, if startFrom ~commit, then create builds with jobs that have requires: ~commit
+ * @method createBuilds
+ * @param  {Pipeline} pipeline                   Pipeline to create builds
+ * @param  {Object}   config
+ * @param  {String}   config.sha                 SHA this project was built on
+ * @param  {String}   config.username            Username of the user that creates this event
+ * @param  {String}   config.scmContext          The scm context to which user belongs
+ * @param  {String}   [config.prRef]             Ref if it's a PR event
+ * @param  {String}   [config.startFrom]         Where the event starts from (jobname or ~commit, ~pr, etc)
+ *                                               Making optional for backward compatibility
+ * @param  {Number}   eventId                    Event's id
+ */
+function createBuilds(pipeline, config, eventId) {
+    const startFrom = config.startFrom;
+
+    // If no startFrom, then do nothing. Remove once we switch to new workflow design
+    if (!startFrom) {
+        return null;
+    }
+
+    // eslint-disable-next-line global-require
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+
+    return pipeline.jobs.then((jobs) => {
```

Comment:
I'm kind of concerned that `createBuilds` is looking up the jobs from the pipeline instead of getting the information from `config.workflow`. I feel like this results in a race condition where the pipeline's workflow could be changed between the event creation and event's initial jobs starting.

Reply:
Sounds like a legit concern. We can change that.